hub/core/chunk.py

```diff
@@ -0,0 +1,117 @@
+from hub.util.exceptions import FullChunkError
+import hub
+from hub.core.storage.cachable import Cachable
+from typing import Sequence, Tuple, Union
+import numpy as np
+from io import BytesIO
+
+from hub.core.meta.encode.shape import ShapeEncoder
+from hub.core.meta.encode.byte_positions import BytePositionsEncoder
+
+
+class Chunk(Cachable):
+    def __init__(
+        self,
+        encoded_shapes: np.ndarray = None,
+        encoded_byte_positions: np.ndarray = None,
+        data: memoryview = None,
+    ):
+        """Blob storage of bytes."""
+
+        self.shapes_encoder = ShapeEncoder(encoded_shapes)
+        self.byte_positions_encoder = BytePositionsEncoder(encoded_byte_positions)
+
+        self._data: Union[bytearray, memoryview] = data or bytearray()
+
+    @property
+    def memoryview_data(self):
+        return memoryview(self._data)
+
+    @property
+    def num_data_bytes(self):
+        return len(self._data)
+
+    def is_under_min_space(self, min_data_bytes_target: int):
+        return self.num_data_bytes < min_data_bytes_target
+
+    def has_space_for(self, num_bytes: int, max_data_bytes: int):
```

Comment:
can't max_data_bytes use the constant?

Reply:
In theory yes, but I believe that chunk max size is configurable. Chunk_engine passes it to this function, which will usually (but not always) equal the default.

Reply:
yeah, @AbhinavTuli you had a TODO in the constants file saying to make configurable, so i made that easier